src/core/opamUrl.ml

```diff
@@ -308,3 +308,21 @@ module Op = struct
     {url with path }
 
 end
+
+module SWHID = struct
+
+  let prefix = "swhid.opam.ocaml.org/"
+  let is_valid u =
+    u.backend = `http
+    && (String.equal u.transport "http" || String.equal u.transport "https")
+    && OpamStd.String.starts_with ~prefix u.path
+  let of_url u =
+    try
+      Some (OpamHash.SWHID.of_string (OpamStd.String.remove_prefix ~prefix u.path))
```

Comment:
IIUC this doesn't support having directly `swh:...` as an URL ? I think we should support both *now* (even if in the beginning we use only `http{s}://swhid.opam.ocaml.org/...` in the official opam repository) to ease the transition to `swh:...` only - which is registered at IANA etc.

Reply:
Tried with simply `swh:...`, it is resolved to a ssh url. There will be another url kind to handle, and besides that, older opams will print the ssh url.